### DISCUSSION TITLE
Refactor redundant selector?

### DISCUSSION THREAD

**Discussion by thian4** [cypress-io/cypress#15238] [Component Testing] [answered]
How to refactor below selector?
```
cy.get('ul.nav').findByText("home").should("exist");
cy.get('ul.nav').findByText("contact").should("exist");
```
I tried to refactor to this
```
const ul = cy.get('ul.nav');
ul.findByText("home").should("exist");
ul.findByText("contact").should("exist"); //failed
```
the second assertion failed.

https://stackoverflow.com/questions/66380511/cypress-passing-selector-to-variable-for-reuse

**Comment by bahmutov** [comment, marked as answer]
You can use either DOM alias [.as](https://on.cypress.io/as) or [.within](https://on.cypress.io/within) command for this. See the full example in https://glebbahmutov.com/cypress-examples/6.5.0/commands/aliasing.html#list-example
